### PR TITLE
Add methods to allow get/set of websocket client ping interval (IDFGH-7096)

### PR DESCRIPTION
**Pull request by 0xFEEDC0DE64**
[Redo of espressif/esp-idf pull 8486](https://github.com/espressif/esp-idf/pull/8486)

### CODE REVIEW

**Comment by gabsuren** [comment]
@0xFEEDC0DE64 thanks for your contribution. The changes LGTM. Some minor nitpicks left.

**Review by gabsuren — CHANGES_REQUESTED** [review]

File: components/esp_websocket_client/esp_websocket_client.c [hunk 1]
```diff
@@ -899,6 +899,37 @@ bool esp_websocket_client_is_connected(esp_websocket_client_handle_t client)
     return client->state == WEBSOCKET_STATE_CONNECTED;
 }
 
+size_t esp_websocket_client_get_ping_interval_sec(esp_websocket_client_handle_t client)
+{
+    if (client == NULL) {
+        ESP_LOGW(TAG, "invalid client");
```
Comment (gabsuren) [inline components/esp_websocket_client/esp_websocket_client.c:905]:
```suggestion
        ESP_LOGW(TAG, "Client was not initialized");
```

File: components/esp_websocket_client/esp_websocket_client.c [hunk 2]
```diff
@@ -899,6 +899,37 @@ bool esp_websocket_client_is_connected(esp_websocket_client_handle_t client)
...
+        return 0;
+    }
+
+    if (client->config == NULL) {
+        ESP_LOGW(TAG, "no config available to change the ping interval for");
```
Comment (gabsuren) [inline components/esp_websocket_client/esp_websocket_client.c:910]:
```suggestion
        ESP_LOGW(TAG, "No config available to change the ping interval");
```

File: components/esp_websocket_client/esp_websocket_client.c [hunk 3]
```diff
@@ -899,6 +899,37 @@ bool esp_websocket_client_is_connected(esp_websocket_client_handle_t client)
...
+}
+
+esp_err_t esp_websocket_client_set_ping_interval_sec(esp_websocket_client_handle_t client, size_t ping_interval_sec)
+{
+    if (client == NULL) {
+        return ESP_ERR_INVALID_ARG;
+    }
+
+    if (client->config == NULL) {
+        ESP_LOGW(TAG, "no config available to change the ping interval for");
```
Comment (gabsuren) [inline components/esp_websocket_client/esp_websocket_client.c:924]:
```suggestion
        ESP_LOGW(TAG, "No config available to change the ping interval");
```

File: components/esp_websocket_client/esp_websocket_client.c [hunk 4]
```diff
@@ -899,6 +899,37 @@ bool esp_websocket_client_is_connected(esp_websocket_client_handle_t client)
...
+        return 0;
+    }
+
+    return client->config->ping_interval_sec;
+}
+
+esp_err_t esp_websocket_client_set_ping_interval_sec(esp_websocket_client_handle_t client, size_t ping_interval_sec)
+{
+    if (client == NULL) {
+        return ESP_ERR_INVALID_ARG;
```
Comment (gabsuren) [inline components/esp_websocket_client/esp_websocket_client.c:920]:
```suggestion
        ESP_LOGW(TAG, "Client was not initialized");
        return ESP_ERR_INVALID_ARG;
```

**Review by gabsuren — CHANGES_REQUESTED** [review]

**Comment by gabsuren** [comment]
@0xFEEDC0DE64 thanks for your contribution. The changes LGTM. Some minor nitpicks left.

**Comment by 0xFEEDC0DE64** [comment]
Did a rebase + squash

**Review by gabsuren — APPROVED** [review]
LGTM!

**Review by david-cermak — APPROVED** [review]

**Pull request merged**